src/UploadInterface.js

```diff
@@ -1,51 +1,161 @@
+
+        const currentCommit = await getCurrentCommit(octokit, username, repository, 'main');
+
+        const newTree = await createNewTree(
+          octokit,
+          username,
+          repository,
+          listOfBlob,
+          pathNames,
+          currentCommit.treeSha
+        );
+
+        setUserMessage('Now creating a commit to repo...');
+        const commitMessage = 'Add my photos to repo';
+        const newCommit = await createNewCommit(
+          octokit,
+          username,
+          repository,
+          commitMessage,
+          newTree.sha,
+          currentCommit.commitSha
+        );
+
+        await setBranchToCommit(octokit, username, repository, 'main', newCommit.sha);
+
+        setUserMessage(
+          `Process complete. You can check out https://github.com/${username}/${repository}`
+        );
+        setLoading(false);
+      } catch (err) {
+        setLoading(false);
+        setErrorOccurred(true);
+        setUserMessage(`Something went wrong while creating a commit`);
+        console.error('Error occurred while creating a commit');
+      }
+    } else {
+      console.error('Error occurred while creating a repo');
+      setUserMessage('Sorry, something went wrong while creating a repository');
+      setErrorOccurred(true);
+      setLoading(false);
+    }
+  }
+
   return (
     <div className="App">
       <header className="preview">
+        {!userMessage ? (
+          <></>
+        ) : loading ? (
+          <Alert variant="info" style={{ width: '85%', textAlign: 'left' }}>
+            <Image src={loadingGif} height="48" width="48" />
+            {userMessage}
+          </Alert>
+        ) : !errorOccurred ? (
+          <Alert variant="success" style={{ width: '85%', textAlign: 'left' }}>
+            {userMessage}
+          </Alert>
+        ) : (
+          <Alert variant="warning" style={{ width: '85%', textAlign: 'left' }}>
+            {userMessage}
+          </Alert>
+        )}
         <img
           ref={imgPreviewEl}
-          className={`img-preview ${!image && "d-none"}`}
-          style={{ display: "block" }}
+          className={`img-preview ${!images && 'd-none'}`}
+          style={{ display: 'block' }}
           alt="Selected file preview"
         />
         <p>Drag and Drop or Click to Upload Images</p>
         <FileUploader
           name="file"
           accept={fileTypes}
           types={fileTypes}
-          multiple={false}
+          multiple={true}
           classes="Drop-zone"
-          handleChange={(file) => {
-            setImage(file);
+          handleChange={(files) => {
+            setImages(files);
             const reader = new FileReader();
             reader.onload = (e) => {
               imgPreviewEl.current.src = e.target.result;
             };
-            reader.readAsDataURL(file);
-            imgPreviewEl.current.title = file.name;
+            reader.readAsDataURL(files[0]);
```

Comment:
Question: on line 137 you allow selecting multiple files, but here assume it's only 1 (i.e., `files[0]`).  Should you be looping through them?

Reply:
If I understand this code correctly, it should read only one image and add it to `imgPreviewEl.current.src` to load the image on the element. If I change it to `files`, it doesn't load the preview on the element. I think it doesn't need to read every file unless if we want to show multiple previews.